### DISCUSSION TITLE
Help Needed: Custom Tooltip Positioning Based on Data Point in LineChart

### DISCUSSION THREAD

**Discussion by Lucc4sMartins** [recharts/recharts#4000] [Q&A] [answered]
I am seeking assistance with customizing the position of tooltips in a `LineChart` component. I want the tooltip to be positioned based on the actual data point, rather than following the mouse cursor. My goal is to have the tooltip statically appear right at the plotted point.

Current behavior (default behavior):
![CleanShot 2023-11-28 at 20 21 53](https://github.com/recharts/recharts/assets/11335135/d9256307-3936-47e5-9f69-30e36d2fb472)

What I wanted it to be (static independent of mouse position):
![CleanShot 2023-11-28 at 20 22 37](https://github.com/recharts/recharts/assets/11335135/2d3000ea-29d7-49ee-983d-7cc50eaa6543)

Here's a Codesandbox with the chart I'm building. Currently, there's nothing going on with the Tooltip as I've forfeited all my attempts to accomplish the goal.

[![Edit Recharts playground (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/recharts-playground-forked-r55k6r)

I feel like I'm really missing something very very silly, would appreciate it very much if someone could shed some light on this 🙏

**Comment by ckifer** [comment]
Hi

You can set the position of the tooltip to a static value if you'd like

```jsx
 <Tooltip
     isAnimationActive={false}
    cursor={false}
    position={
      { y: 50 }
    }
/>
```

[![Edit Recharts playground (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/recharts-playground-forked-jdtczz?file=%2Findex.js)

**Reply by ckifer** [reply]
This storybook example is also helpful 
https://github.com/recharts/recharts/blob/master/storybook/stories/Examples/Tooltip.stories.tsx#L71

https://master--63da8268a0da9970db6992aa.chromatic.com/?path=/story/examples-tooltip--locked-by-click

**Reply by Lucc4sMartins** [reply]
Thank you for jumping in so quickly!

Sorry, I haven't done a good job describing what I've tried and why I failed...

I've tried `position`, but it seems to be tied to the `CartesianGrid` instead of the data points -- at least that's what I feel as the tooltip position is always set to the same height when setting, per se, `{ y: 50 }`.

---

I wanted it to be relative to the data points, in a way they would always have the same distance from the data points. Something like this:

![CleanShot 2023-11-28 at 21 09 41](https://github.com/recharts/recharts/assets/11335135/932e8977-a7e9-4ab7-9ea3-4a577017aacc)
![CleanShot 2023-11-28 at 21 10 06](https://github.com/recharts/recharts/assets/11335135/4efeeb96-539c-4b39-97e5-5e078111ab5f)


I'm now exploring if the example given in `Locked by click` is helpful in this case, thanks for pointing to that.

[off-topic] A quick feedback on the Storybook: I thought it wasn't working because it took too long to load from the homepage, it'd be amazing to have visual feedback it's loading all the components.

**Reply by ckifer** [reply, marked as answer]
Makes sense - yeah its tied to the grid with 0, 0 being the top left corner.

See this issue https://github.com/recharts/recharts/issues/488 - a lot of "workarounds" there, but currently this isn't implemented in an easy way in recharts. We don't have enough help to take a ton of changes on at a time so it hasn't been prioritized. If you'd like to help it would be an amazing way to contribute!

>[off-topic] A quick feedback on the Storybook: I thought it wasn't working because it took too long to load from the homepage, it'd be amazing to have visual feedback it's loading all the components.

This is on storybook, I don't think we have any control over the loading state of things unfortunately - it should show a loading bar previous to loading the content.

**Reply by Lucc4sMartins** [reply]
https://github.com/recharts/recharts/issues/488 gave me some great great insights on how to do it and I found my way to accomplish it.

Thank you so much for your guidance and kindness!

**Reply by ckifer** [reply]
Glad you were able to! Feel free to at least share here or in that thread so others can benefit too!

**Reply by Lucc4sMartins** [reply]
It was a combination of some existing responses, not sure I can add much to it.

Once I'm finished, I'll make sure to double-check if there's room for contribution in that sense!

**Reply by filsonfp** [reply]
@Lucc4sMartins would you like to share some thoughts about your solution? 😉 I'm wondering if there is any clean solution possible to achieve that, without dumping some nice features from recharts (like curson on line chart)

**Reply by ckifer** [reply]
This code is not quite "clean" but someone recently implemented something over in this issue https://github.com/recharts/recharts/issues/4919#issuecomment-2306763550 (see sandbox link there) 

As I mention there and above there isn't quite a clean solution today unfortunately. Hopefully soon it'll be easier

**Reply by NarendraKuruva** [reply]
@filsonfp 
I tried it this way. And it worked in my use case

[Here is the codesandbox link ](https://codesandbox.io/p/sandbox/recharts-tooltip-follow-datapoint-forked-f2qrd2?file=%2Fsrc%2FApp.tsx%3A6%2C9)
